src/Microsoft.AspNetCore.Mvc.Razor/DependencyInjection/MvcRazorMvcCoreBuilderExtensions.cs

```diff
@@ -151,27 +152,33 @@ internal static void AddRazorViewEngineServices(IServiceCollection services)
             //
             // Razor compilation infrastructure
             //
-            services.TryAddSingleton<RazorProject, FileProviderRazorProject>();
-            services.TryAddSingleton<RazorTemplateEngine, MvcRazorTemplateEngine>();
```

Comment:
These are breaking changes. I think we need to leave it as-is.

Reply:
OH nvm, I see you just moved it. Can you log an issue to remove these in 3.0?

Reply:
https://github.com/aspnet/Mvc/issues/7381